pulsar-zookeeper-utils/src/test/java/com/yahoo/pulsar/zookeeper/ZookeeperCacheTest.java

```diff
@@ -449,4 +449,30 @@ public String deserialize(String key, byte[] content) throws Exception {
         scheduledExecutor.shutdown();
 
     }
+
+    /**
+     * Test to ensure that the cache puts on watch even on nodes that do not yet exist.
+     */
+    @Test
+    public void testExistsWatch() throws Exception {
+        ExecutorService zkExecutor = Executors.newSingleThreadExecutor(new DefaultThreadFactory("mockZk"));
```

Comment:
Make sure to shutdown the 3 executors or we are leaking threads in the test execution